Source/JavaScriptCore/heap/Heap.cpp

```diff
@@ -1753,7 +1755,7 @@ NEVER_INLINE bool Heap::runEndPhase(GCConductor conn)
 
     // updateAllocationLimits() updates this for us.
     if (overCriticalMemoryThreshold())
-        sweepSynchronously();
+        m_needSynchronousSweep = true;
```

Comment:
Nit: You could probably just remove this and ask if memory is critical in `shouldSweepSynchronously`